scratch/simulation.cc

```diff
@@ -0,0 +1,309 @@
+    for (uint32_t i = 0; i < num_nodes; i++) 
+    {
+        for (uint32_t j = 0; j < num_nodes; j++)
+        {
+            if (i != j)
+            {
+                ApplicationContainer apps;
+
+                //
+                // Create a BulkSendApplication and install it on node i.
+                //
+                uint16_t port = 9000 + i;
+
+                BulkSendHelper source("ns3::TcpSocketFactory", InetSocketAddress(interface.GetAddress(j), port));
+                source.SetAttribute("MaxBytes", UintegerValue(0));
+                apps.Add(source.Install(nodes.Get(i)));
+
+                // Create a callback, which indicates when the TCP connection succeeds, so can set
+                // up traces.
+                DynamicCast<BulkSendApplication>(apps.Get(0))->SetConnectCallback(&traceHandler);
+
+                //
+                // Create a PacketSinkApplication and install it on node 1
+                //
+                PacketSinkHelper sink("ns3::TcpSocketFactory", InetSocketAddress(Ipv4Address::GetAny(), port));
+                apps.Add(sink.Install(nodes.Get(j)));
+                apps.Start(Seconds(0.0));
+                apps.Stop(Seconds(5000.0));
+
+                matrix[i][j] = apps;
+            }  
+        }
+    }
+
+
+    //
+    // Set up tracing on L2
+    //
+    AsciiTraceHelper ascii;
+    Ptr<OutputStreamWrapper> stream = ascii.CreateFileStream("l2.tr");
+    p2p.EnableAscii(stream, devices.Get(0));
+
+
+    //
+    // The next block of code is for testing because I've been unable to read from
+    // the flow file. See below for explanation of issue.
+    //
+
+    // Fake input for testing between 2 nodes and schedule them.
+    std::vector<std::vector<std::string>> input;
+    std::string input_data[] = {
+        "1 0 3 100 50773 2.000227265",  
+        "0 1 3 100 3929948 2.001185171",  
+        "1 0 3 100 30071 2.002157443",  
+        "1 0 3 100 676 2.003654232",  
+        "0 1 3 100 76409 2.005367012",  
+        "0 1 3 100 1649191 2.009730850",  
+        "0 1 3 100 8993 2.009783566",  
+        "1 0 3 100 15700100 2.011084367",  
+        "1 0 3 100 20305770 2.012322672",  
+        "1 0 3 100 59025 2.013073333"
+    };
+    for (const std::string& line : input_data) 
+    {
+        std::vector<std::string> params;
+        std::istringstream iss(line);
+        std::string token;
+        while (std::getline(iss, token, ' ')) 
+        {
+            params.push_back(token);
+        }
+        Simulator::Schedule(Seconds(std::stod(params[5])), 
+                            &flowHandler, 
+                            matrix[std::stoi(params[0])][std::stoi(params[1])], 
+                            std::stoi(params[4])); 
+    }
+
+    //
+    // TODO Schedule flows based on file input. It's unable to read from the file because
+    // it immediately hits an end of file?
+    // 
+
+    // std::ifstream file;
+    // file.open("tmp_traffic.txt");
```

Comment:
```
std::ifstream file;
file.open("~/home/david/ns-3/scratch/tmp_traffic.txt");
NS_ASSERT(!file.is_open());
NS_ASSERT(!file.good());
NS_ASSERT(file.eof());
```

I've tried it with absolute path, and in both cases, the third assertion fails; meaning that it's able to open the file,  but it immediately sees and eof, which is definitely not the case. The same problem exists even if I give it a different file to read.

Reply:
Try seeking to the beginning of the file. Maybe the file open() mode is incorrect.

Reply:
I did some more debugging, and I found its a issue with ns3 build system. Whenever I compile and run with gcc, the file stream works, but it doesn't work when I build with ns3. I've tried cleaning and re-building, and the problem persists. I think it may be something with CMake?